\newacronym{DNA}{DNA}{deoxyribonucleic acid}
\newacronym{T1D}{T1D}{type 1 diabetes}
\newacronym{IL-2}{IL-2}{interleukin-2}
\newacronym{MZ}{MZ}{monozygotic}
\newacronym{DZ}{DZ}{dizygotic}
\newacronym{PBMC}{PBMC}{peripheral blood mononuclear cells}

%disease
\newacronym{AML}{AML}{actute myeloid leukemia}

% genetic polymorphism
\newacronym{SNP}{SNP}{single nucleotide polymorphism}
\newacronym{CN}{CN}{copy number}
\newacronym{CNV}{CNV}{copy number variation}
\newacronym{HWE}{HWE}{Hardy-Weinberg equilibrium}
\newacronym{LD}{LD}{linkage disequilibrium}

\newacronym{KIR}{KIR}{killer immunoglobulin-like receptor}
\newacronym{NK}{NK}{natural killer}
\newacronym{HLA}{HLA}{human leukocyte antigen}
\newacronym{ITIM}{ITIM}{immune tyrosine-based inhibitory motif}

% resources
\newacronym{1958BC}{1958BC}{British 1958 Birth Cohort}
\newacronym{GRID}{GRID}{Genetic Resource Investigating Diabetes}
% newly diagnosed and unaffected siblings
\newacronym{D-GAP}{D-GAP}{Diabetes - Genes, Autoimmunity and Prevention}

% biological methods
\newacronym[\glslongpluralkey={genome-wide association studies},\glsshortpluralkey={GWAS}]{GWAS}{GWAS}{genome-wide association study}
\newacronym{qPCR}{qPCR}{quantitative Polymerase Chain Reaction}

%\newacronym{DILT1D}{DILT1D}{adaptive study of IL-2 dose on regulatory T cells in type 1 diabetes}

% stats methods
\newacronym{AIC}{AIC}{Akaike Information Criterion}
\newacronym{ANOVA}{ANOVA}{analysis of variance}
\newacronym{BIC}{BIC}{Bayesian Information Criterion}
\newacronym{EM}{EM}{expectation maximisation}
\newacronym{MCMC}{MCMC}{Markov Chain Monte Carlo}
\newacronym{knn}{KNN}{k-nearest neigbour}
\newacronym{GMM}{GMM}{Gaussian mixture model}
\newacronym{LOOCV}{LOOCV}{leave-one-out cross-validation}
\newacronym{ANN}{ANN}{approximate-nearest-neighbour}
\newacronym{CART}{CART}{classification and regression tree}
\newacronym{RF}{RF}{random forests}
\newacronym{MARS}{MARS}{multivariate adaptive regression splines}
\newacronym{PRIM}{PRIM}{Patient Rule Induction Method}
\newacronym{MDS}{MDS}{multidimensional scaling}
\newacronym{MST}{MST}{minimum spanning tree}
\newacronym{RMSE}{RMSE}{root mean square error}
\newacronym{MSD}{MSD}{mean square difference}
\newacronym{MSE}{MSE}{mean square error}
\newacronym{MISE}{MISE}{mean integrated square error}
\newacronym{RSS}{RSS}{residual sum of squares}
\newacronym{SPADE}{SPADE}{Spanning-tree Progression of Density-normalised Events}
\newacronym{PCA}{PCA}{principal component analysis}
\newacronym{ISOMAP}{ISOMAP}{isometric feature mapping}

%

% flow cytometry
\newacronym{FCS}{FCS}{Flow Cytometry Standard}
\newacronym{HIP}{HIP}{Human Immunology Project}
\newacronym{MFI}{MFI}{mean fluorescence intensity}
\newacronym{MEF}{MEF}{molecules of equivalent fluorochrome}
\newacronym{APC}{APC}{allophycocyanin} 
\newacronym{FlowCAP}{FlowCAP}{Flow Cytometry Critical Assessment of Population Identification Methods}

%genes
\newacronym{IL2RA}{IL2RA}{interleukin-2 receptor alpha}
\newacronym{PTPN2}{PTPN2}{protein tyrosine phosphatase N2}
\newacronym{FOXP3}{FOXP3}{forkhead box P3}
\newacronym{STAT5}{STAT5}{signal transducer and activator of transcription 5}
