\newacronym{DNA}{DNA}{deoxyribonucleic acid}
\newacronym{T1D}{T1D}{type 1 diabetes}
\newacronym{IL-2}{IL-2}{interleukin-2}
\newacronym{MZ}{MZ}{monozygotic}
\newacronym{DZ}{DZ}{dizygotic}
% genetic polymorphism
\newacronym{SNP}{SNP}{single nucleotide polymorphism}
\newacronym{CN}{CN}{copy number}
\newacronym{CNV}{CNV}{copy number variation}
\newacronym{HWE}{HWE}{Hardy Weinberg equilibrium}
\newacronym{LD}{LD}{linkage disequilibrium}

\newacronym{KIR}{KIR}{killer immunoglobulin-like receptor}
\newacronym{NK}{NK}{natural killer}
\newacronym{HLA}{HLA}{human leukocyte antigen}
\newacronym{ITIM}{ITIM}{immune tyrosine-based inhibitory motif}
\newacronym{PCA}{PCA}{principal component analysis}
% resources
\newacronym{1958BC}{1958BC}{British 1958 Birth Cohort}
\newacronym{GRID}{GRID}{Genetic Resource Investigating Diabetes}
% newly diagnosed and unaffected siblings
\newacronym{D-GAP}{D-GAP}{Diabetes: Genes, auto-immunity and prevention}
% biological methods
\newacronym[\glslongpluralkey={genome wide association studies},\glsshortpluralkey={GWAS}]{GWAS}{GWAS}{genome wide association study}
\newacronym{qPCR}{qPCR}{quantitative Polymerase Chain Reaction}
% stats methods
\newacronym{ANOVA}{ANOVA}{analysis of variance}
\newacronym{EM}{EM}{expectation maximisation}
\newacronym{knn}{KNN}{k-nearest neigbour}
\newacronym{GMM}{GMM}{Gaussian mixture model}
\newacronym{LOOCV}{LOOCV}{leave-one-out cross-validation}
\newacronym{ANN}{ANN}{approximate-nearest-neighbour}
\newacronym{CART}{CART}{classification and regression tree}
\newacronym{RF}{RF}{random forests}
\newacronym{MARS}{MARS}{multivariate adaptive regression splines}
\newacronym{MST}{MST}{minimum spanning tree}
%

% flow cytometry
\newacronym{HIP}{HIP}{Human Immunology Project}
\newacronym{MFI}{MFI}{mean fluorescence intensity}
\newacronym{MEF}{MEF}{molecules of equivalent fluorochrome}
\newacronym{APC}{APC}{allophycocyanin}

