\newacronym{DNA}{DNA}{deoxyribonucleic acid}
\newacronym{T1D}{T1D}{Type 1 Diabetes}
\newacronym{IL-2}{IL-2}{interleukin-2}
\newacronym{MZ}{MZ}{monozygotic}
\newacronym{DZ}{DZ}{dizygotic}
\newacronym{PBMC}{PBMC}{Peripheral Blood Mononuclear Cells}

%disease
\newacronym{AML}{AML}{Acute Myeloid Leukemia}

% genetic polymorphism
\newacronym{SNP}{SNP}{Single Nucleotide Polymorphism}
\newacronym{CN}{CN}{Copy Number}
\newacronym{CNV}{CNV}{Copy Number Variation}
\newacronym{HWE}{HWE}{Hardy-Weinberg Equilibrium}
\newacronym{LD}{LD}{Linkage Disequilibrium}

\newacronym{KIR}{KIR}{Killer Immunoglobulin-like Receptor}
\newacronym{NK}{NK}{Natural Killer}
\newacronym{HLA}{HLA}{Human Leukocyte Antigen}
\newacronym{ITIM}{ITIM}{Immune Tyrosine-based Inhibitory Motif}

%cells
\newacronym{Treg}{Treg}{regulatory T cell}
\newacronym{Teff}{Teff}{effector T cell}

% resources
\newacronym{1958BC}{1958BC}{British 1958 Birth Cohort}
\newacronym{GRID}{GRID}{Genetic Resource Investigating Diabetes}
% newly diagnosed and unaffected siblings
\newacronym{D-GAP}{D-GAP}{Diabetes - Genes, Autoimmunity and Prevention}

% biological methods
\newacronym[\glslongpluralkey={Genome-Wide Association Studies},\glsshortpluralkey={GWAS}]{GWAS}{GWAS}{Genome-Wide Association Study}
\newacronym{qPCR}{qPCR}{quantitative Polymerase Chain Reaction}

%\newacronym{DILT1D}{DILT1D}{adaptive study of IL-2 dose on regulatory T cells in type 1 diabetes}

% stats methods
\newacronym{ML}{ML}{Maximum Likelihood}
\newacronym{AIC}{AIC}{Akaike Information Criterion}
\newacronym{ANOVA}{ANOVA}{ANalysis Of VAriance}
\newacronym{BIC}{BIC}{Bayesian Information Criterion}
\newacronym{EM}{EM}{Expectation Maximisation}
\newacronym{MCMC}{MCMC}{Markov Chain Monte Carlo}
\newacronym{knn}{KNN}{k-nearest neigbour}
\newacronym{GMM}{GMM}{Gaussian Mixture Model}
\newacronym{LOOCV}{LOOCV}{leave-one-out cross-validation}
\newacronym{LOO}{LOO}{leave-one-out}
\newacronym{ANN}{ANN}{Approximate-Nearest-Neighbour}
\newacronym{CART}{CART}{Classification And Regression Tree}
\newacronym{RF}{RF}{Random Forests}
\newacronym{MARS}{MARS}{Multivariate Adaptive Regression Splines}
\newacronym{PRIM}{PRIM}{Patient Rule Induction Method}
\newacronym{MDS}{MDS}{multidimensional scaling}
\newacronym{MST}{MST}{Minimum Spanning Tree}
\newacronym{RMSE}{RMSE}{Root Mean Square Error}
\newacronym{MSD}{MSD}{Mean Square Difference}
\newacronym{MSE}{MSE}{Mean Square Error}
\newacronym{MISE}{MISE}{Mean Integrated Square Error}
\newacronym{RSS}{RSS}{Residual Sum of Squares}
\newacronym{SPADE}{SPADE}{Spanning-tree Progression of Density-normalised Events}
\newacronym{PCA}{PCA}{Principal Component Analysis}
\newacronym{PLS}{PLS}{Partial Least Squares}
\newacronym{ISOMAP}{ISOMAP}{isometric feature mapping}
\newacronym{viSNE}{viSNE}{visual Stochastic Neighbour Embedding}

%

% flow cytometry
\newacronym{PMT}{PMT}{Photo Multipler Tube}
\newacronym{FCS}{FCS}{Flow Cytometry Standard}
\newacronym{HIP}{HIP}{Human Immunology Project}
\newacronym{MFI}{MFI}{Mean Fluorescence Intensity}
\newacronym{MEF}{MEF}{Molecules of Equivalent Fluorochrome}
\newacronym{APC}{APC}{allophycocyanin} 
\newacronym{FlowCAP}{FlowCAP}{Flow Cytometry Critical Assessment of Population Identification Methods}
\newacronym{SPICE}{SPICE}{Simplified Presentation of Incredibly Complex Evaluations}

%genes
\newacronym{IL2RA}{IL2RA}{interleukin-2 receptor alpha}
\newacronym{PTPN2}{PTPN2}{protein tyrosine phosphatase N2}
\newacronym{FOXP3}{FOXP3}{forkhead box P3}
\newacronym{STAT5}{STAT5}{signal transducer and activator of transcription 5}
