\chapter{ \label{chapter:il2} Methods to assess cell response to ex-vivo stimulation in flow cytometry }
%\chapter{ \label{chapter:il2} Effect of IL-2 Stimulation on Cell Phenotypes } 
%\chapter{ \label{chapter:il2} A recursive partitioning approach of identifying cells responsive to ex-vivo stimulation in flow cytometry }

In the previous chapter, we looked at normalisation of the MFI using beads and replicating two univariate gating steps with thresholding on CD25
or with a mixture of two components on CD45RA.
Here on a different dataset on which only preliminary manual analysis has been done, we will also consider normalisation methods and replicating all steps of the manual gating
with an automatic procedure.
Furthermore, we will also consider automatic approaches for discovering biologically relevant subsets not reported by the manual gating.

\section{Background}

\paragraph{Motivation}
%\Acrfull{GWAS}
Genomewide association studies have implicated the \cytokine{IL-2} signalling as an important aetiological pathway associated with the development of \gls{T1D}.  
%So far associated regions have been reported close to IL2RA and PTPN2, a phosphatase, which both influence STAT5 transduction.
%The protective rs12722495 haplotype was significantly associated with increase in CD25 expression on CD4 memory T cells measured in terms of MFI.
%Howevever the protective combined rs2104286 genotype led to a decrease in the percentage of CD25 positive CD4 naive T cells.  
As seen in \Cref{chapter:il2ra}, the protective T1D associated \gene{IL2RA} variant at \snp{rs12722495} in healthy individuals predicts an increase in expression of \protein{CD25},
the alpha chain of the trimeric \protein{IL-2} receptor, on memory \protein{CD4}\positive T lymphocytes \citep{Dendrou:2008gc,Dendrou:2009dv}.
%and high levels of IL-2 secretion by these cells
%As supported by an increase in soluble CD25, this might be the consequence of preferential cleavage of the CD25 receptor \citet{Lowe:2007ij}.
%As supported by a decerase in soluble CD25 \citet{Lowe:2007ij}.
\citet{Garg:2012jr} found that regulatory and memory \protein{CD4}\positive T cells in healthy carriers of T1D risk associated \gene{IL2RA} variants,
also exhibit decreased sensitivity to \cytokine{IL-2} in terms of decreased MFI levels of
phosphorylated signal transducer and activator of transcription 5 protein (p\protein{STAT5}),
STAT5 dimerises or tetramerises on phosphorylation and acts as
a transcription factor (\Cref{figure:jones-2000}),
which also induces the transcription of \gene{FOXP3}, a transcription factor characteristic of regulatory T cells .

\begin{figure}[h]
\centering
\includegraphics[scale=0.75]{IL2/figures/IL2-pathway.jpg}
\mycaption{figure:jones-2000}
{Schematic of major IL-2 signaling pathway (taken from \citet{Liao:2013jt}).}
{
\protein{CD25} is \protein{IL2RA}, the $\alpha$ subunit of the trimeric \cytokine{IL-2} receptor.
The $\alpha$ chain is the highest binding affinity receptor of the three chains.
STAT5 is phosphorylated to pSTAT5 and acts as a transcription factor.
%Schematic of Major IL-2 Signaling Pathways
%Shown is the activation of PI 3-K-AKT, JAK-STAT, and SHC-RAS-MAPK signaling pathways. Also shown are potential therapeutic points of control for IL-2 signaling, with anti-Tac (daclizumab), rapamycin, and JAK1 or JAK3 inhibitors being shown in red. The cartoon shows signaling by both STATs dimers and tetramers. The figure indicates that IL-2 activates more STAT5 than STAT3 and more STAT3 than STAT1. ERK refers to both ERK1 and ERK2. MEK refers to both MEK1 and MEK2.
}
\end{figure}

\citet{Long:2011hk} have reported that a \Gls{T1D} associated variant at \snp{rs1893217}
of the protein tyrosine phosphatase N2 gene (\gene{PTPN2}),
a negative regulator of the IL-2 pathway,
also correlates with lower STAT5 phosphorylation.
%show diminished ability to respond to \cytokine{IL-2}.
Furthermore, it is suspected that \cytokine{IL-2} production might be diminished in T1D since disease associated \gene{IL2RA} variants 
correlate with reduced CD25 levels 
and reduced \cytokine{IL-2} production
on activated CD69\positive CD4\positive memory T cells
after antigen stimulation \citep{Dendrou:2009dv}.
%with staphylococcal enterotoxin B 
%really?
Long-term reduced sensitivity to IL-2 also correlates with dimished maintenance 
of FOXP3 expression in the CD4\positive CD25\positive regulatory T-cells of type 1 diabetic subjects \citep{Long:2010ej}.
%Defects in IL-2R signaling contribute to diminished maintenance 
%IL-2 sensitivity drives IL-2 stimulation
%Preliminary data also suggests that the sensitivity to IL-2, measured in terms of pSTAT5 activation, may be also be reduced in the regulatory CD4+ T cells of newly diagnosed T1D patients.
%Furthermore it is also suspected that IL-2 production might be diminished in T1D
%since disease associated variants in the \emph{IL2RA} and \emph{PTPN2} genotypes correlate with reduced IL-2 production (reference?). 
Hence, these findings appear to consolidate the hypothesis that type 1 diabetics tend to have a reduced ability to respond to IL-2
in part due to genetic defects in \gene{PTPN2}, \gene{IL2RA} and possibly other gene variants involved in the IL-2 signalling pathway
%and that this decreased sensitivity is most noticeable in regulatory T cells
\citep{Long:2010ej,Long:2011hk,Long:2012ea}.
%The cell subsets are not always consistently defined across studies.
%Long:2010ej define Tregs as CD4+ CD25hi (FOXP3 not measured)
%Long:2011hk report that CD4+ CD45RO+
%nor is the definition of response to IL-2
%sometimes pSTAT5 MFI is measured, sometimes pSTAT5 pct positive, sometimes pSTAT5 fold change
These results are of great clinical relevance to us,
since our lab is currently conducting an
adaptive study of IL-2 dose on regulatory T cells in type 1 diabetes (\url{http://www.clinical-trials-type1-diabetes.com/}),
and to all researchers interested in low-dose IL-2 therapy to autoimmune diseases \citep{Koreth:2011kv,Saadoun:2011em}.

However some concerns have been raised with these studies.
One concern was that the Tregs discrimination was not particulary thorough.
%nor consistent
\citet{Long:2010ej} define Tregs based only on two markers CD4\positive CD25\positive whereas these cells are usually
also defined on CD127 and FOXP3.
Another a notable omission by \citet{Long:2010ej} was the lack of repeated samples to assess the within-individual variance
or reliability of the assay.
%Although \citet{Long:2011hk} do assess repeatability in 16 individuals, its not accounted for in the association test.
%clinical trials which
%treat type 1 diabetics with IL-2.
Hence, \contributor{Tony Cutler}, in our lab, set to find if he could replicate some of these findings in an independent cohort
using a more refined gating stategy by including the FOXP3 regulatory T cell marker,
%T1D leads to decrease sensitivity to IL-2 in similar sample size with 
as well as NK cell markers, CD3, CD8, and CD56,
%and the naive cell marker CD31,
to discover whether other potential cell subsets
are also sensitive to IL-2 (\Cref{table:IL2-panels}).
%and using repeated samples.

\paragraph{Samples and panels}

He selected 22 long-standing diabetics (6 males and 16 females, mean age 29) and 28 controls (mean age 27) from the Cambridge Bioresource,
as well as 30 newly diagnosed (20 males and 9 females, mean age 11.7) and 15 unaffected siblings (5 males and 12 females, mean age 12.3) from the \Gls{D-GAP} resource.  
%This was to determine if there was a difference between newly diagnosed and long-standing diabetics while controlling for age.
To guard from false positive association and non-reproducible results, 
it is important to ascertain the repeatability of these cell phenotypes before conducting any form of association study.
%Before proceeding with association testing of these cell phenotypes with T1D status or other covariates,
In order to test the repeatability, ten individuals were recalled for a second blood sample (\Cref{table:IL2-recalled-individuals}).
Hence a total of 52 cases and 43 controls, of which, 10 individuals (5 cases and 5 controls) from the Cambridge Bioresource were recalled to assess reproducibility
of the phenotypes.  

\begin{table}[ht]
\centering
\begin{tabular}{llll}
  \hline
Individual & status  & pch & number of days between visits \\
  \hline
1          & control & a   & 98 \\
2          & case    & b   & 140 \\
3          & control & c   & 167 \\
4          & control & d   & 98 \\
5          & case    & e   & 167 \\
6          & case    & f   & 112 \\
7          & control & g   & 112 \\
8          & case    & h   & 98 \\
9          & case    & i   & 120 \\
10         & control & j   & 140 \\
   \hline
\end{tabular}
\mycaption{table:IL2-recalled-individuals}
{Ten individuals recalled between 98 and 168 days later to assess stability of the cell phenotypes. }
{
pch is the plotting character used to refer to these individuals in plots later in this chapter.
}
\end{table}
%In order to test these hypotheses, we selected 21 long-standing diabetics and 30 controls from the Cambridge Bioresource,
%as well as  28 newly diagnosed and 14 unaffected siblings from DGAP.
%The nature of the resource allows more flexibility to recall patients of interest, i.e those with low IL-2 signalling potential, for more in depth analysis.
%To test this hypothesis, blood samples are taken from 21 type 1 diabetics and 30 healthy controls.  
Blood samples were prepared and analysed by flow cytometry on day of collection.
Each sample was split into four aliquots of \SI{500}{\micro\litre}.
The first aliquot was left unstimulated.
The remaining three were stimulated ex-vivo for 30 minutes at four 100-fold increasing doses
of proleukin (a polymer of \cytokine{IL-2}) at $0.1$, $10$ and \SI{1000}{\unit\per\milli\litre} respectively.
%\SI{1000}{\unit\per\micro\litre} respectively.
After a set stimulation time, the samples were fixed, permeabilised and stained, with different panels (\Cref{table:IL2-panels}), 
on a set of core markers, not expected to be affected by short-term proleukin stimulation,
CD4, CD25, CD45RA and FOXP3.
These were used to delineate different cell types, and the functional marker, pSTAT5
, a signalling protein of the IL-2 pathway phosphorylated on IL-2 stimulation (\Cref{figure:jones-2000}),
was used to measure IL-2 response.
Samples were analysed individually with flow cytometry, with all \gls{T1D} samples except for two run in parallel with healthy controls
to account for batch effects (\Cref{figure:IL2-sample-time}).
Samples were also matched for age and sex to allow for paired analysis.

%We address the following research questions relating to the pSTAT5 response in these samples:
%\begin{itemise}
  %%\item Can some individuals be classified as low/high responders?
  %\item In the cell subsets identified by manual gating, is the response to IL-2 associated with case-control status or any other covariate? 
  %\item Using automated methods, can we find other responsive cell subsets which are ignored by the manual gating?
%\end{itemise}
%Answering the first question will depend on the reproducibility of the response between cell-subsets within individuals,
%and we will see if, as in \Cref{chapter:il2ra}, normalisation can improve the reproducibility.

%Tony Cutler, Saleha Hassan, Marcin Pekalski will be involved in immune function assays.
%Louise Bell will be ethics liason.
%Statistical analysis will be carried out by a delegated member of the Statistics team.
%Demands on DIL infrastructure 
%Blood handling rooms, BD LSRII Fortessa machine.  
\begin{table}[h!]\footnotesize
  \centering
\begin{tabularx}{\textwidth}{lcccc}
\rowcolor{Gray}
Flurochrome       & T/NK cell panel & CD4 T cell panel & CD4/naive cell panel & NK cell panel \\
Alexa Fluor 488   & pSTAT5          & pSTAT5           & pSTAT5               & pSTAT5  \\
Alexa Fluor 700   & CD4             & CD4              & CD4                  & CD4     \\
APC               & CD25            & CD25             & CD25                 & CD25    \\
Pacific Blue      & CD56            & CD45RA           & CD45RA               & CD45RA  \\
PE YG             & FOXP3           & FOXP3            & FOXP3                & FOXP3   \\
PE-Cy7 YG         & CD45RA          &                  & CD31                 & CD56    \\
PerCP Cy5-5       & CD3             &                  &                      & \\
Qdot 605          & CD8             &                  &                      & \\
\hline
Number of samples & 26              & 95               & 12                   & 66 \\
%Number of samples & 13 & 96 &
\end{tabularx}
\mycaption{table:IL2-panels}
{Proleukin stimulation assay antibody-fluorochrome panels.}
{
The fluorochrome-antibody panels used in IL-2 stimulation.
The panel used on the majority of samples was the CD4 T cell panel, used to disriminate
effector and regulatory naive and memory T cells.
The T/NK cell panel, which contains the most markers, was only ran on a subset of 26 samples.
}
\end{table}
%CD14-CD19-CD3-CD45RA-CD56-HLA-DR-PSTAT5
%1
%CD19-CD25-CD27-CD3-CD45RA-CD56-CD8-PSTAT5
%1
%CD19-CD25-CD27-CD3-CTLA4-PSTAT5
%1
%CD19-CD25-CD27-CD3-ISO-PSTAT5
%1
%CD25-CD31-CD4-CD45RA-FOXP3-PSTAT5
%5
%CD25-CD3-CD4-CD45RA-CD56-CD8-CD8-PSTAT5
%1
%CD25-CD3-CD4-CD45RA-CD56-CD8-FOXP3-PSTAT5
%13
%CD25-CD3-CD4-CD45RA-CD56-CD8-PSTAT5
%48
%CD25-CD3-CD4-CD45RA-CD56-FOXP3-PSTAT5
%37
%CD25-CD4-CD45RA-CD56-FOXP3-PSTAT5
%5
%CD25-CD4-CD45RA-FOXP3-KI67-PSTAT5
%1
%CD25-CD4-CD45RA-FOXP3-PSTAT5
%50
%
%\begin{figure}
%\centering
%\includegraphics[scale=.5] {flowdatasets/figures/il2-stimulation-samples-time.pdf}
%\mycaption{figure:IL2-sample-time} 
%{ Days on which samples are processed. }
%{
  %To account for day of analysis effect in the case-control association testing,
  %at least one healthy control sample and one type 1 diabetic sample were analysed per day.
%}
%\end{figure}

\myfigure{scale=.5}{IL2-sample-time} 
{ Number of cases and controls analysed per day. }
{
  To account for batch effects in the case-control association testing,
  Tony Cutler aimed to analyse, when possible,
  at least one healthy control sample and one type 1 diabetic sample per day.
  %However on some days the number of cases and controls did not match.
  On two days, no matching controls were ran.
}




\section{Cell Phenotypes Identified by Manual Analysis}

In the preliminary manual analysis using FlowJo, \contributor{Tony Cutler},
%The pSTAT5 distribution was measured in effector and regulatory naive and memory T cell populations (Figure~\ref{figure:tony-cd4-gating}).
gated four CD4\positive lymphocyte subsets (\Cref{figure:tony-cd4-gating}):
\begin{itemise}
  \item memory effector
  \item memory treg
  \item naive effector
  \item naive treg
\end{itemise}
%\begin{figure}[h]
    %\centering
    %\includegraphics[scale=.75]{IL2/figures/tony-cd4-gating.pdf}
    %\caption{  \label{figure:tony-cd4-gating} Manual gating conducted by \contributor{Tony Cutler} to identify:
    %conventional and regulatory naive and memory T cells as well as activated regulatory T cells.
    %}
%\end{figure}
% 
%\hspace{-2cm}
\begin{figure}[h]
\centering
%\includegraphics[scale=.5]{figures/manual-gating.pdf}
%\begin{subfigure}[b]{.4\textwidth}
%\includegraphics[scale=.25]{IL2/figures/manual-gating-CB00165D_0U_2012-11-29.pdf}
  \includegraphics[scale=.75]{IL2/figures/CB00366X_2012-11-07.pdf}
%\caption{Resting sample.}
%\end{subfigure}
%\begin{subfigure}[b]{.4\textwidth}
%\includegraphics[scale=.25]{IL2/figures/manual-gating-CB00165D_01U_2012-11-29.pdf}
%\caption{Stimulated at 0.1 units of proleukin.}
%\end{subfigure}
%\begin{subfigure}[b]{.4\textwidth}
%\includegraphics[scale=.25]{IL2/figures/manual-gating-CB00165D_10U_2012-11-29.pdf}
%\caption{Stimulated at 10 units of proleukin.}
%\end{subfigure}
%\begin{subfigure}[b]{.4\textwidth}
%\includegraphics[scale=.25]{IL2/figures/manual-gating-CB00165D_1000U_2012-11-29.pdf}
%\caption{Stimulated at 1000 units of proleukin.}
%\end{subfigure}
\mycaption{figure:tony-cd4-gating}
{Gates applied across doses.}
{
Manual gating conducted using FlowJo by \contributor{Tony Cutler} to identify
naive effector (green ellipse) and regulatory T cells (blue ellipse) (e)
and memory effector (black ellipse) and regulatory T cells (red ellipse) (f).
%The same gates can be applied across doses (i, ii, ,iii, iv).
}
\end{figure}
Within each lymphocyte subset, the pSTAT5 distribution was measured, in each of the four samples stimulated at an increasing proleukin dose
(\Cref{figure:dose-effect-pstat5-cellsubsets}).  
As expected, the pSTAT5 distribution shifts progressively right for higher doses of proleukin, as more STAT5 is phosphorylated.
Of the four subsets, the most sensitive cells to proleukin are the smaller memory and naive regulatory T cell subsets (\Cref{figure:dose-effect-pstat5-cellsubsets} b and d),
then the memory effectors (\Cref{figure:dose-effect-pstat5-cellsubsets} a) and finally the naive effectors (\Cref{figure:dose-effect-pstat5-cellsubsets} c).
This correlates with the level of \protein{CD25} expressed by these cells.

%\includegraphics[scale=.45]{IL2/figures/dose-effect-pstat5-cellsubsets-density.pdf}
%\includegraphics[scale=.45]{IL2/figures/dose-effect-pstat5-cellsubsets-density-baseline.pdf}
%\includegraphics[scale=.45]{figures/dose-effect-pstat5-cellsubsets-ecdf-joined.pdf}
\myfigure{scale=.6}
%individual g
{dose-effect-pstat5-cellsubsets-density}
{ Distribution pSTAT5 in the manually gated cell subsets in the sample manually gated in \Cref{figure:tony-cd4-gating}. }
{
The thickness of the lines is representative of the four increasing doses of proleukin (0, 0.1, 10 and 1000 units).
The dose-response is most striking in the smaller Treg subsets with higher CD25 (b and d).
The dashed vertical line represents the 99th percentile of the pSTAT5 distribution in the resting sample,
which is used to define the pSTAT5\positive threshold.
}

Next \contributor{Tony Cutler}, assessed the repeatability of the pSTAT5 response by measuring the pSTAT5 \gls{MFI}
for each individual subset of lymphocytes tested.
However, this cell phenotype was poorly reproducible, since the location of the peaks was not stable across days as illustrated
by the sample shown in \Cref{figure:dose-effect-pstat5-cellsubsets-density-repeatability}.
%Furthermore, the \gls{MFI} is not a representative metric for bimodal distributions 
%However, one issue is that pSTAT5 MFI is not very representative for very bimodal or skewed distributions such
%as the memory effectors in \Cref{figure:dose-effect-pstat5-cellsubsets}.
%Also the resting pSTAT5, which represents the baseline pSTAT5, of the same cell subset may not be constant across days.  
%To address these issues, the percent of pSTAT5\positive cells was also assessed.
\myfigure{scale=.6}
%individual g
{dose-effect-pstat5-cellsubsets-density-repeatability}
{ pSTAT5 distribution in an individual on visit one (black) and visit two (red) clearly shows that pSTAT5 distribution is not stable across days in the four cell
subsets. }
{
In black, the pSTAT distribution on visit one and in red, on visit two.
%The thickness of the line represents the proleukin dose at which the sample was stimulated (0, 0.1, 10 and 1000 units).
The thinner lines are from the resting sample whereas the thicker lines are from the sample stimulated at 1000 units.
The vertical lines represent the pSTAT5\positive threshold set at the $99^{th}$ percentile of the pSTAT5 distribution in the resting sample.
}
This motivated \contributor{Tony Cutler} to instead use a threshold approach to define the ratio of cells which are pSTAT5\positive.
An idea similar to that applied in \Cref{chapter:il2ra} to define naive cells as CD25\positive.
%but the threshold is defined using the unstimulated sample rather than beads. 
%It is a good approach to use when the number of cells is small
%and is commonly used in populations with high CV.
However, here an internal threshold was used, namely
the threshold was defined to be the 99th percentile of the pSTAT5 distribution in the resting cell subset per sample.
Thus each each sample and cell subset had its own pSTAT5\positive threshold.
He presented his results in six of the ten repeated individuals for the four stimulated cell subsets,
memory effector (\Cref{figure:tony-memory-eff}),
memory regulatory T cells (\Cref{figure:tony-memory-treg}),
naive effector (\Cref{figure:tony-naive-eff})
and naive regulatory T cells (\Cref{figure:tony-naive-treg}).
For memory and naive Tregs (\Cref{figure:tony-memory-treg,figure:tony-naive-treg}), at the highest 1000 units dose,
practically all memory and naive tregs are pSTAT5\positive.
However, these cell populations show a significant response already at the lowest 0.1 units dose of proleukin.
Based on this observation, \contributor{Tony Cutler} selected the
memory and naive Treg pSTAT5 cell phenotype to be the percentage of pSTAT5\positive cells at the 0.1 units dose.
On the other hand, since memory effectors are less reponsive, 10 units was chosen as the representative dose.
For naive effectors, the least reponsive of the four cell subsets considered, the repeatability was assessed at the 1000 unit dose.
Tony assessed the repeatability with the coefficient of determination:
%instead of using the square of the Pearson correlation: 
\[
  R^2 = 1 - \frac{\sum_{i=1}^N (x_{i1}-x_{i2})^2}{\sum_{i=1}^N (x_{i1}-\overline{x_1})^2}
\]
where $x_{i1}$ is the phenotype of $i^{th}$ individual on the first day and $x_{i2}$
is the phenotype of the $i^{th}$ individual on the second day.
The coefficient of determination can take negative values if the correlation between $x_{i1}$ and $x_{i2}$ is very low.
Contrary to the Pearson correlation, used in the previous chapter, the coefficient of determination is sensitive to linear transforms.
%this is not correct because it's not symmetric
Even when using the percent of pSTAT5\positive cell phenotype,
the overall reproducibility across the four cell subsets was still poor,
with the more sensitive and smaller cell subsets,
naive and memory Tregs showing the worst correlation ($R^2=-0.16$ and $R^2=-0.82$), memory effectors showing slightly better correlation ($R^2=0.021$)
and finally the less responsive naive effectors showing good correlation ($R^2=0.7728$) (\Cref{figure:tony-repeatability}).
Tony then went on to test association with type 1 diabetes using a two-tailed paired t-test of 20 cases matched with 20 controls analysed on the same
day (\Cref{figure:tony-t1d-association}).
%how did he deal with repeats?
He also tested for association with \gene{IL2RA} SNP \snp{rs12722495}, and the
\gene{PTPN2} SNPs \snp{rs45450798} and \snp{rs478582} (plots not shown).
No significant association was detected either with disease nor with genotype.

\myfigure{scale=.5}
{tony-memory-eff}
{ The percent of pSTAT5\positive cells increases with proleukin dose in memory effectors,
but the measured response is not consistently repeatable (f, g). }
{
  Plot produced by Tony Cutler.
}
\myfigure{scale=.5}
{tony-memory-treg}
{ The percent of pSTAT5\positive cells increases with proleukin dose in memory tregs. }
{
  Plot produced by Tony Cutler.
  While at the highest proleukin dose of 10 and 1000 units, all memory tregs are consistently pSTAT5\positive,
  there is more discrepancy at the low dose of 0.1 units.
}
\myfigure{scale=.5}
{tony-naive-eff}
{ The percent pSTAT5\positive cells increases with proleukin dose in naive effectors. }
{
  Plot produced by Tony Cutler.
  Only \pct{40} of the naive effector cells are pSTAT5\positive even at the highest 1000 unit proleukin dose.
}
\myfigure{scale=.5}
{tony-naive-treg}
{ The percent pSTAT5\positive cells increases with proleukin dose in naive tregs. }
{
  Plot produced by Tony Cutler.
  While at the highest proleukin doses of 10 and 1000 units, all naive tregs are consistently pSTAT5\positive,
  there is more discrepancy at the low dose of 0.1 units.
}
\myfigure{scale=.75}
{tony-repeatability}
{ Repeatability in six individuals }
{
  Plot produced by Tony Cutler.
  The repeatability of the percent of cells which are pSTAT5\positive is assessed in effector memory and naive at the 10U and 1000U dose
  respectively, and in memory and naive regulatory T cells at the 0.1U dose.
  While the repeatability in the naive effector subset was good ($R^2=0.7728$), the repeatability in the other cell subsets is poor
  (memory effectors $R^2=0.021$, naive tregs $R^2=-0.16$ and memory tregs $R^2=-0.82$).

}
\myfigure{scale=.5}
{tony-t1d-association}
{ Association test of percent pSTAT5\positive in four cell subsets. }
{
  Plot produced by Tony Cutler.
  The association with T1D of the percent of cells which are pSTAT5\positive is assessed in effector memory and naive at the 10U and 1000U dose
  respectively, and in memory and naive regulatory T cells at the 0.1U dose.
  The association test is a two-tailed paired-t-test on 20 cases paired with 20 controls analysed on the same day (40 out of the available 96 individuals).
  No significant association is detected.
}

%\section{Automatic methods}
%\subsection{Normalisation of pSTAT5 improves repeatability}
\section{Reproducibility of pSTAT5 response within an individual}

Tony's preliminary results in the six repeated individuals suggested that the pSTAT5 MFI and percent pSTAT5\positive were poorly reproducible
cell phenotypes using the methodology and approach described, which consequently would give us little power to detect an association with disease status or genetics.
This motivated me to see if I could improve the repeatability of these cell phenotypes using normalisation.
Furthermore, since these cell phenotypes were only defined at a single dose,
I also defined new cell phenotypes to summarise the response across doses.
The reproducibility of all these cell phenotypes was tested with different normalisation approaches.


\subsection{pSTAT5 response cell phenotypes}

The sensitivity of the response in the four manually gated cell subsets is confirmed by plotting proleukin dose against pSTAT5 MFI (\Cref{figure:pstat5-mfi-auc-celltypes}.a).
It also illustrates that pSTAT5 saturation occurs in naive and memory regulatory T cells at 10 units of proleukin while naive effector cells do not reach
saturation even at the highest dose of 1000 units.

%Provided the pSTAT5 distribution is unimodal then a shift in the peak of the pSTAT5 density should correspond to a difference in the MFI,
%however this does not hold if the distribution is not unimodal as seen for the naive effectors (\Cref{figure:dose-effect-pstat5-cellsubsets}).

An alternative metric which captures the response of the whole distribution across all doses instead of single dose, 
is the area between the pSTAT5 cumulative density functions in the resting and stimulated samples.
This relates to the Kolmogorov-Smirnoff distance which is defined as the maximum vertical distance between cumulative density functions,
but also captures the horizontal dimension.

%The repeatability of these metrics was tested but I found none to be repeatable neither in the gated subsets nor any of the larger subsets.
%(\Cref{XXX}).
%Instead I decided to use the area under the pSTAT5 dose-response curve for each cell type (\Cref{figure:pstat5-auc-celltypes}).

%The repeatability of pSTAT5 MFI at all doses in each cell subset was tested using the manual gating and my method which allowed for
%the gates to move so as to recenter on the data (\Cref{figure:repeatability-gating}).
%It was found that overall the pSTAT5 MFI was poorly repeatable using both gating approaches,
%with the most sensitive cells to proleukin, showing the worse repeatabilty in pSTAT5 MFI.
%To capture the response across all days, the sum of the pSTAT5 MFI was used.
%and not just the portion which crosses the positivity treshold,
%There are two established metrics for measuring the distance between distributions:
%the Kullback-Leibler (KL) and Kolomogorov-Smirnoff (KS) distance.
%the KL distance is not symmetric

%The relative shift in the pSTAT5 distribution between resting and stimulated can be measured by computing the area between the cumulative density functions of the unstimulated and stimulated samples.
% As the pSTAT5 signal increases in a stimulated sample it is expected that the CDF shifts right.
% The resting provides the baseline.
%The area between the pSTAT5 CDF in the resting sample and the pSTAT5 CDF in the stimulated sample is measure of the strength of the response.
%Ten of these low responders from the Cambridge Bioresource were recalled for re-analysis to assess stability of this phenotype.
%However this cell phenotype was not found to be reproducible.  
%Using the automatically adjusted gates from manual gating, we assess the repeatability of the response in the four identified cell subsets.
%Defining a threshold for pSTAT5 positivity is done using a similar approach as in \Cref{chapter:il2ra}.
%The top percentile of the pSTAT5 distribution in the resting sample is used as a threshold as described in %\Cref{ }.  
%Some individuals with a reduced pSTAT5 response consistent across studied cell types were identified by Tony, however there was no evidence of association with disease status.  
%There are unsolved day to day variation issues, better to do in bulk from frozen once Foxp3 IC stain works
%A question of potential interest with whether the low responders show any other immunological phenotype such as for example Th17, IL-21 production, Tfh proportion?

%Normalisation can rely on peak alignment of the ungated sample but the peaks are often hard to reliably identify.
%The fluorescent marker required for pSTAT5 is Alexa Fluor 488
%Absolute MFI is a not a reliable way to assess pSTAT5.
%We need to subtract the background mfi.
%We can either do this on a cluster basis or use ann joining.
%What is more repeatable?  

\subsection{Normalisation approaches}

I attempted to use normalisation to improve the repeatability of the proleukin response measure.
Here I describe the methods I applied.

\paragraph{Bead normalisation} 
The normalisation method applied to CD25 normalisation,
in \Cref{chapter:il2ra}, was to use beads to correct for day variation.
However for the stimulation data, beads in the Alexa-488 channel, the fluorochrome conjugated to pSTAT5 (\Cref{table:IL2-panels}),
%\Cref{chapter:il2ra}
did not adequately capture the short term variation in pSTAT5 (\Cref{figure:pstat5-beads}).  
This suggests that for this experiment, flurorescent beads do not explain the day to day variation in the pSTAT5 MFI.
\begin{figure}[h]
    \centering
    %\includegraphics[scale=.5]{IL2/figures/beads.pdf}
    %\includegraphics[scale=.5]{IL2/figures/beads.pdf}
    %\includegraphics[scale=.75]{figures/alexa488-beads.pdf}
    \includegraphics[scale=.6]{figures/pstat5-beads.pdf}
    \mycaption{figure:pstat5-beads}
    {Variation in sample MFI is not captured by variation in bead MFI.}
    {
      For the purpose of MFI normalisation, the fluorescence intensity of six-peak flow cytometry beads
      was measured in the Alexa 488 channel, the fluorochrome conjugated to the pSTAT5 marker.
      However, as illustrated by the loess lines, the beads are not appropriate for pSTAT5 normalisation,
      since the MFI of the three dimmest populations of beads (orange)
      does not capture the pSTAT5 MFI time variation in the four cell subsets.
      The pSTAT5 MFIs are obtained from samples stimulated at 1000 units of proleukin.
      
    }
\end{figure} 


\paragraph{ Subtracting baseline MFI }
One observation which can be made from \Cref{figure:dose-effect-pstat5-cellsubsets-density-repeatability} is that the baseline
is different across days.
If a cell population had a higher resting pSTAT5 MFI then one might expect that the pSTAT5 MFI in the stimulated population would also be higher.
%First I noticed that Tony had not corrected for baseline MFI.
I first attempted to account for the difference in pSTAT5 baseline by subtracting the MFI of the resting population from the MFI of
the population in the sample at the three increasing doses.
However, this does not appear to improve the repeatability significantly but instead just reduces the MFI by the same factor on both days.
This suggests that the day to day variation in the resting sample MFI does not explain the variation in the stimulated population
(\Cref{figure:pstat5-mfi-cellsubsets-repeatability}).
\myfigure{scale=1}{pstat5-mfi-cellsubsets-repeatability}
{ pSTAT5 MFI (black), background subtracted pSTAT5 MFI (red) }
{
  Correcting for the MFI in the resting sample does not appear to improve the repeatability significantly but instead just reducing the MFI by
  the same factor on both days.
  This suggests that the day to day variation in the resting sample MFI does not explain the variation in the stimulated population.
}


\paragraph{Peak alignment}
%The repeatability of the pSTAT MFI was assessed.  
Another approach to the normalisation as implemented in the \Rpackage{flowStats}, is to align the modes
of the distributions across days in the ungated sample.
The method assumes that the two peaks in the pSTAT5 distribution,
representing stimulated and unstimulated cells,
should exist in all ungated samples, but that their locations and their proportions may vary across doses and days.
%Given that the location of the peaks of pSTAT5 in the ungated sample remains fairly stable across doses but that the heights vary,
%this advocates a normalisation method which relies on peak alignment.
%If we expect that the core cell populations should exist in all samples but that their locations and their proportions may vary across days and dose,
%a normalisation approach is to align the location of the population across doses per day.
%Hence we wish to align the location of the peaks across days while allowing for the relative proportion of the populations to vary.  
%The idea is to identify the groups based on a single marker by clustering algorithm and then selecting the peaks to be the group medians.
%In this approach the number of groups K needs to be known a priori and is assumed to be the same across samples.
%This is the method used by \Rpackage{flowBeads} which identifies groups with the k-medoids algorithm for the purpose of bead normalisation.
%In Chapter 3, I used k-means to identify the peaks, but I found on this data k-means and k-medoids performed poorly since the cluster variances are very different.  
However peaks are not always clearly identifiable.
In particular, the second peak corresponding to the stimulated cells is much broader and smaller since the majority of cells in the sample are not stimulated.
%A mixture model algorithm would be better suited since it allows for different variances in the clustering.
%but because of the skewness of the data, the peaks did not correspond to the cluster medians.
To identify the peaks, I used a sliding window approach on the density function.
The sliding window approach records the point with the highest density estimate in the current window,
so that within a window of twice the span, the point of highest density is selected as a peak.
At the end, a list of highest density points is returned of which the top K may be selected.
I found in practice this method worked well with a window span of $40$ across all samples considered.
%This approach is useful for estimating K but the window size may need to be tuned per sample.  
Unfortunately in flow cytometry, the peaks are not always clearly identifiable due to noise associated with staining.
This is particularly noteable in ungated data where debris obfuscate the pSTAT5 signal and can introduce artefactual peaks, leading to peak misidentification.
Furthermore, the transform can also introduce spurious peaks close to zero.
%depending on the value of the width parameter selected for the logicle transform.
%(\Cref{appendix:transformation,appendix:normalisation}).
However I found that upon gating on lymphocytes and CD4\positive lymphocytes, these spurious peaks disappear and both peaks were correctly identified 
(\Cref{figure:pstat5-peak-normalisation}).
\begin{figure}[h]
    \centering
    \includegraphics[scale=.5]{IL2/figures/pstat5-peak-normalisation.pdf}
    \mycaption{figure:pstat5-peak-normalisation}
    {Normalisation by peak alignment of the pSTAT5 distribution in lymphocytes (a and b) and CD4\positive lymphocytes (c and d).}
    {
      The two peaks of the pSTAT5 distribution are identified using a sliding window approach,
      in the lymphocyte sample stimulated at 1000 units (a) and in the manually gated CD4\positive lymphocyte subset stimulated at 10 units (c).
      In the normalised pSTAT5 distribution in the ungated (a) and CD4\positive lymphocyte subset (d),
      the identified peaks in (a) and (c) have been mapped to the respective medians across all peaks.
    }
\end{figure} 
In the CD4\positive lymphocytes, in order to facilitate peak identification, I considered the pSTAT5 distribution at 10 units instead of 1000 units,
as the unstimulated/stimulated ratio is more even at an intermediate dose.
%since at the highest dose, most CD4\positive lymphocytes are stimulated.
%Here's an example of this method applied on simulated data where the number of peaks is known and easily identifiable (Figure~\ref{figure:simulation-peak-align}).
%The sliding window method is good for detecting local maximums while the clustering approach is better at find global splits.
%Hence we combine both approaches by using the sliding window on the partitions identified by the clustering.
%We apply mclust and pam (clara) because mclust tends to the fit the data better when the cluster are of very uneven size like example (~/pam.fail.RDAta).
Once the two peaks were identified in all samples,
%the next task is now to align for the purpose of normalisation.
%A non-linear transform could be considered so that peaks may be aligned independently, but
a linear transformation was used to exactly map the two peaks across all samples, as shown in \Cref{figure:pstat5-peak-normalisation}.
%since there are only two peaks, a linear transform is sufficient 
%After applying this transform to the previously gated pSTAT5 MFI and retested the repeatability.
%We also apply this method the pSTAT5 positivity threshold
%how?  

\paragraph{Nearest-neighbour joining}
%While using a different background MFI should account for differences in resting MFI,
%this assumes that the MFI difference is representative of the pSTAT5 shift.
%This may not be true for distributions which change modality.
One concern is that if cell counts are very different across stimulation doses, this could influence the accuracy of the MFI estimate.
Instead, a more correct approach would be to subtract the background for each cell.
One way of emulating this is to match each cell to its closest neighbour in the unstimulated sample.  
This was accomplished by first joining samples on their core markers using the \gls{ANN} to the resting sample \citep{Jones:2011ez}
as implemented in the \Rpackage{RANN}.
This created a dataset of the same number of cells as the resting dataset, but where each cell now had a total of four functional pSTAT5 markers,
one for each stimulation dose.
At each cell it is now possible to assess the difference in pSTAT5 response between resting and stimulated states.
This is important because cells do not all have the same resting level of pSTAT5 (\Cref{figure:pstat5-baseline-relative}).
This approach presents a number of advantages.
Firstly, only the sample to which the other samples are joined needs to be gated.
Secondly, since the pSTAT5 response is relative to the baseline, it should be more robust to variation between days
and consequently, more reproducible than pSTAT5 fluorescence intensity.
Thirdly, since we have response at the cell-level, we can use methods
such as \gls{CART}, \gls{RF} and \gls{MARS} to do multivariate regression of pSTAT5 from core markers.
This could help identify cells which would have been missed from only examining core markers.
%However, even when we correct for the different baseline, the distance between the peak of the resting sample and the stimulated sample is not constant
%figure
%An advantage of this method is that number of cells per cell type is constant across doses.
%A disadvantage of this method is that the number of cells is very different of that all the cells are shifted then they will all be joined to closest neighbour
%which could be a single point.
%which will investigate in the second section.
%An alternative to this approach, would be to gate subsets in subsamples stimulated at different doses and subtract the background,
%however a clear advantage of this approach is that it allows for methods such as RF and MARS
The \gls{ANN} approach is valid without normalisation since the distributions of the core markers align across doses.
%(\Cref{figure:core-markers}).
\Cref{figure:ann-join-0U-01U} confirms that the mapping of the clusters is correct across the nearest-neighbour mapping.
%When the between-dose variation on the core markers is negligeable,
%If we assume that the within-tube variation is negligeable, then a simpler solution is possible:
%each cell in the unstimulated subsample is joined to its closest neighbouring cell (in terms of its core markers) in a stimulated subsample.
%This assumes that the core markers are fairly stable across samples, which is usually the case in ex-vivo stimulated.
%The joining results is also not affected by any euclidean distance preserving transform, so the FCS data can be transformed before or after the \gls{ANN} joining.
%Another important check is to see whether the joining is sensitive to the transform.
%We find that we get the same results whether we join before or after the transform.
%\myfigure{scale=.3}{dose-effect-core-markers}{Dose-effect on core markers in ungated sample.} { Increasing doses represented by lines of increasing thickness.  The distributions align well across doses.  }
%\hspace{-2cm}
\begin{figure}[h]
\centering
\includegraphics[scale=.3]{IL2/figures/ann-join-0U-01U.pdf}
\mycaption{figure:ann-join-0U-01U}
{Nearest neighbour join on core markers of ungated resting sample with ungated sample stimulated at 0.1 units. }
{
All clusters lie on the y=x line which suggests that the largest clusters are correctly matched across samples.
}
\end{figure} 
%\hspace{-2cm}
\begin{figure}[h]
\centering
\includegraphics[scale=.5]{IL2/figures/pstat5-baseline-relative.pdf}
\mycaption{figure:pstat5-baseline-relative}
{ pSTAT5 intensity across the four proleukin doses, before (a) and after (b) per-cell baseline pSTAT5 subtraction in the ungated sample.}
{
  An important proportion of cells are already saturated for pSTAT5 (high baseline) in the resting sample (a).
  Correcting for the per cell pSTAT5 baseline, shows the true proportion of cells which responds to proleukin within this sample (b).
}
\end{figure}
\myfigure{scale=.5}{pstat5-mfi-auc-celltypes}
{ pSTAT5 MFI for each normalisation method.  }
{
  The solid lines represent the median.
  The shaded areas represent the 25th and 75th percentile.
  The doses at which most of the variation occurs are 1000 units for naive effectors (green),
  10 units for memory effectors (black), 0.1 units for memory (red) and naive (blue) regulatory T cells.
  %a) The dose-response curve for each cell type in all samples.
  %b) The medians dose-response curve across all samples for each cell type.
}
\myfigure{scale=.5}{pstat5-pos-auc-celltypes}
{ Percent pSTAT5\positive for each normalisation method. }
{
  The solid lines represent the median.
  The shaded areas represent the 25th and 75th percentile.
  The doses at which most of the variation occurs are 1000 units for naive effectors (green),
  10 units for memory effectors (black), 0.1 units for memory and naive regulatory T cells.
  %a) The dose-response curve for each cell type in all samples.
  %b) The medians dose-response curve across all samples for each cell type.
  %There are many samples which stand out as outliers and a few of them have been misgated.
}
%\myfigure{scale=.5}{pstat5-auc-boxplots-celltypes.pdf} { Influence of normalisation method on pSTAT5 area under curve. } { In red the normalisation using the peak alignment in CD4\positive lymphocytes, in blue the normalisation using peak alignment in lymphocytes. }
%\begin{figure}[h]
    %\centering
    %\includegraphics[scale=.5]{IL2/figures/pstat5-auc-repeatability-celltypes.pdf}
    %\mycaption{figure:pstat5-auc-repeatability-celltypes}
    %{ Influence of normalisation on the repeatability of the pSTAT5 area under the curve. }
    %{ In red the normalisation using the peak alignment in CD4\positive lymphocytes, in blue the normalisation using peak alignment in lymphocytes. }
%\end{figure}
\myfigure{scale=.75}
{nn-pstat5-mfi-cellsubsets-repeatability}
{ Nearest-neighbour joined pSTAT5 MFI (black), NN background subtracted pSTAT5 MFI (red) }
{ 
  Background subtraction does no substantially improve the repeatability of the pSTAT5 MFI phenotype.
}




\subsection{Repeatability}


The NN joining does not significantly improve the repeatability (\Cref{figure:nn-pstat5-mfi-cellsubsets-repeatability}).

For all ten repeated samples, the Pearson correlation squared $r^2$ of the MFI was assessed at the four increasing doses, in the four cell subsets,
for the raw (\Cref{figure:repeatability-PSTAT5-MFI}a),
baseline corrected (\Cref{figure:repeatability-PSTAT5-MFI}b),
nearest-neighbour joined (\Cref{figure:repeatability-PSTAT5-MFI}c)
and nearest-neighbour baseline corrected (\Cref{figure:repeatability-PSTAT5-MFI}d).
The repeatability of the percent pSTAT5\positive across doses gives a different pattern depending on the cell type
but at the highest dose it appears that the least responsive naive effector cells yield the best
repeatability (\Cref{figure:repeatability-PSTAT5-pos}).
Suprisingly, the memory regulatory T cells have the highest repeatability at the 0.1 unit dose.
The repeatability of the naive regulatory T cell phenotype is poor at all doses.
No normalisation approach stands out as improving the repeatability.

\myfigure{scale=.5}
{repeatability-PSTAT5-MFI}
{
  Repeatability of pSTAT5 MFI measured as Pearson correlation squared ($r^2$) per dose per cell type.
}
{
  On subtracting the baseline, the repeatability of the naive effector subset is improved but not in the other cell subsets.
} 
\myfigure{scale=.5}
{repeatability-PSTAT5-pos}
{
  Repeatability of the percent of pSTAT5\positive as Pearson correlation squared ($r^2$) per dose per cell type.
}
{
  %While the repeatability of the percent pSTAT5\positive is more reproducible.
}

%
%\hspace{-2cm}
%\begin{figure}[h]
%\centering
%\includegraphics[scale=.4]{IL2/figures/pstat5-response-cellsubsets-manualgates.pdf}
%\caption{ \label{figure:pstat5-response-cellsubsets-gate}
%Fixed gates (a and b), magnetic gates (c and d).
%The response is measured in terms of baseline relative pSTAT5 MFI.
%a) The dose-response per cell-type in each sample obtained from manual gating with fixed gates.
%Note the outstanding memory effector sample is due to poor gating of the memory populatin in that particular sample.
%%CB01500E
%This in part due to the gate position but also because that sample does not have a clear memory population.
%b) The median dose-response per cell-type obtained from manual gating with fixed gates.
%}
%\end{figure}
%

%\section{pSTAT5 response metrics}
%\section{pSTAT5 response in the manually gated cell subsets} 
%\section{Joining samples on core markers using approximate nearest-neighbour}

%\hspace{-2cm}
%\begin{figure}[h]
%\centering
%%\includegraphics[scale=.5]{figures/manual-gating.pdf}
%\begin{subfigure}[b]{.4\textwidth}
  %\includegraphics[scale=.25]{IL2/figures/repeatability-pstat5-mfi-Memory-Eff.pdf}
%\caption{Memory Effector}
%\end{subfigure}
%\begin{subfigure}[b]{.4\textwidth}
  %\includegraphics[scale=.25]{IL2/figures/repeatability-pstat5-mfi-Memory-Treg.pdf}
%\caption{Memory Treg}
%\end{subfigure}
%\begin{subfigure}[b]{.4\textwidth}
  %\includegraphics[scale=.25]{IL2/figures/repeatability-pstat5-mfi-Naive-Eff.pdf}
%\caption{Naive Effector}
%\end{subfigure}
%\begin{subfigure}[b]{.4\textwidth}
  %\includegraphics[scale=.25]{IL2/figures/repeatability-pstat5-mfi-Naive-Treg.pdf}
%\caption{Naive Treg}
%\end{subfigure}
%\caption{
%\label{figure:repeatability-gating}
%In black manual gates, in red magnetic gates.
%The AUC is the area under the dose response curve.
%}
%\end{figure}
%

%In the manual analysis conducted by Tony Cutler, he found that he could identify individuals which were low or high responders.
%This leads us to hypothesise that within one individual the response to \emph{IL-2} on day 1 should be comparable to the response to \emph{IL-2} on day 2.

%One method we have tried is based on the assumption is that in the ungated data, the bottom and top percentile of the pSTAT5 distribution do not respond to \emph{IL-2} and so can be used as reference points.  This is effectively a quantile normalisation method aligning the bottom and top percentile across doses.

%However we found that this normalisation method d is not true in CD4+ lymphocyte gated data.  
%Certain normalisation methods make assumptions about the shape of the data.
%The actual choice of normalisation depends on the characteristics of the data we wish to compare.
%Certain normalisation methods such as quantile normalisation are not appropriate since they assume the distributions have the shame shape.


%\begin{figure}[h]
%%
%%#simulated data
%%x0 <- mixtools::rnormmix(1000, mu=c(1,6), lambda=c(.3,.7))
%%x1 <- mixtools::rnormmix(1000, mu=.9*c(1,6)+2, lambda=c(.5,.5), sigma=c(2,2))
%%d0 <- density(x0)
%%d1 <- density(x1)
%%l0 <- extract.landmarks(x0,max.lms=2, bw=d0$bw)
%%l1 <- extract.landmarks(x1,max.lms=2, bw=d1$bw)
%%m <- lm(l0$lms ~ l1$lms)
%%x1.norm <- cbind(1,x1)%*%coefficients(m)
%%l1.norm <- extract.landmarks(x1.norm,max.lms=2)
%%#before align
%%pdf('~nikolas/GoogleDrive/PhD/Thesis/IL2/figures/simulation-peak-align-noise.pdf')
%%plot(d0,xlim=c(-3,11),main='', xlab='')
%%points(l0$lms, l0$dens, pch=20, cex=2)
%%lines(d1, lty=2)
%%points(l1$lms, l1$dens, pch=20, cex=2)
%%#after peak align
%%lines(density(x1.norm), lty=2, col='red')
%%points(l1.norm$lms, l1.norm$dens, pch=20, cex=2, col='red')
%%dev.off()
%%
    %\centering
    %\includegraphics[scale=1]{IL2/figures/simulation-peak-align-noise.pdf}
    %\caption{  \label{figure:simulation-peak-align-noise}  In solid black line represents the density function obtained from $1000$ draws from a mixture of two normal distribution
    %with means $\mu_0=(1,6)$, standard deviations $\sigma_0=(1,1)$ and mixing proportions $\tau_0=(.3,.7)$.
    %The dashed black line represents the density function obtained from $1000$ draws from a mixture of two normal distribution
    %where $\mu_1 = 0.9 \mu_0 + 2$, standard deviations $\sigma_0=(1,1)$ and $\tau_1=(.5,.5)$.
    %The red dashed line represents the transformation using peak alignment. }
%\end{figure}
%

%%e <- lapply(flat.rep.fcs[names(flat.rep.fcs)[1:4]], function(x) ecdf(lgcl(getChannels(x, 'pstat5'))))
%%d <- lapply(flat.rep.fcs[names(flat.rep.fcs)[1:4]], function(x) density(lgcl(getChannels(x, 'pstat5')),bw=.1))
%%d.f <- lapply(d, function(d) splinefun(d$x,d$y))
%%y.max <- max(sapply(d, function(d) d$y)) 
%%e <- lapply(lymph[names(lymph)[1:4]], function(x) ecdf(lgcl(getChannels(x, 'pstat5'))))
%%d <- lapply(lymph[names(lymph)[1:4]], function(x) density(lgcl(getChannels(x, 'pstat5')),bw=.1))
%%d.f <- lapply(d, function(d) splinefun(d$x,d$y))
%%y.max <- max(sapply(d, function(d) d$y)) 
%%#pdf('~nikolas/lymph-dose-effect.pdf',width=10,height=5)
%%pdf('~nikolas/ungated-dose-effect.pdf',width=10,height=5)
%%par(mfrow=c(1,2))
%%x <- seq(-.5, 3, length.out=20000)
%%plot(x, e[[1]](x), col='white', xlab='pSTAT5', xlim=c(-.5,3), ylab='') 
%%mapply(function(e,lwd,lty) lines(x,e(x),lwd=lwd,lty=lty),e,seq(1,2.5,.5),c(1,2,2,1))
%%legend('topleft',doses, lwd=seq(1,2.5,.5), lty=c(1,2,2,1))
%%plot(x, d.f[[1]](x), col='white', xlab='pSTAT5', xlim=c(-.5,3),ylim=c(0,1),ylab='') 
%%mapply(function(d,lwd,lty) lines(x,d(x),lwd=lwd,lty=lty),d.f,seq(1,2.5,.5),c(1,2,2,1))
%%legend('topleft',doses, lwd=seq(1,2.5,.5), lty=c(1,2,2,1))
%%dev.off() 
%%pdf('~nikolas/ungated-lymph-dose-effect.pdf')
%%par(mfrow=c(1,1))
%%x <- seq(-.5, 3, length.out=20000)
%%plot(x, d.f[[1]](x), col='white', xlab='pSTAT5', xlim=c(-.5,3),ylim=c(0,1),ylab='') 
%%d <- lapply(flat.rep.fcs[names(flat.rep.fcs)[c(1,4)]], function(x) density(lgcl(getChannels(x, 'pstat5')),bw=.1))
%%d.f <- lapply(d, function(d) splinefun(d$x,d$y))
%%y.max <- max(sapply(d, function(d) d$y))
%%mapply(function(d,lwd,lty) lines(x,d(x),lwd=lwd,lty=lty),d.f,c(1,2.5),1)
%%d <- lapply(lymph[names(lymph)[c(1,4)]], function(x) density(lgcl(getChannels(x, 'pstat5')),bw=.1))
%%d.f <- lapply(d, function(d) splinefun(d$x,d$y))
%%y.max <- max(sapply(d, function(d) d$y))
%%#r <- mapply(function(a,b) length(a)/length(b), lymph[1:4], flat.rep.fcs[1:4])
%%mapply(function(d,lwd,lty) lines(x,d(x),lwd=lwd,lty=lty,col='red'),d.f,c(1,2.5),1) 
%%legend('topleft',doses[c(1,4)], lwd=c(1,2.5), lty=1)
%%dev.off()
%%lymph <- sapply(flat.rep.fcs, function(x) { cd4 <- lgcl(getChannels(x, 'cd4')); return(x[2 <  cd4 & cd4 < 2.75,]) })
%%

\subsection{Association of pSTAT5 response with type 1 diabetes}

We tested the association with T1D at each dose as well as for the area under the curve.
Having improved the reproducibility of the cell phenotype with normalisation via peak alignment,

we are now in a position to test
the original hypothesis of whether type 1 diabetics show a reduced response to proleukin in those cell subsets.
We account for repeated individuals with a linear mixed effects model as applied in Chapter 2.%\Cref{chapter:il2ra}.
%\contributor{Hui Guo} also attempted a stratified boostrapping to compute the variance for a paired-test
Using the paired design, we also control for sex and day of analysis.

Having not identified a significant association, this now brings us to the next question of whether there are other
cell subsets besides the ones which we are gating which are sensitive to proleukin, and in particular low doses of proleukin?
%
%\begin{figure}[h]
    %\centering
    %\includegraphics[scale=.5]{IL2/figures/pstat5-auc-t1d-celltypes.pdf}
    %\mycaption{figure:pstat5-auc-t1d-celltypes}
    %{ Influence of normalisation on the association of pSTAT5 area under the curve with type 1 diabetes in the four cell subsets. }
    %{ }
%\end{figure}

% pSTAT5 MFI
\myfigure{scale=.5}{pstat5-mfi-t1d-celltypes}
{ Association test of percent pSTAT5\positive with disease status. }
{ }
\myfigure{scale=.5}{nn-pstat5-mfi-t1d-celltypes}
{ Association test of percent pSTAT5\positive with disease status. }
{ }

% pct pSTAT5+
\myfigure{scale=.5}{pstat5-pos-t1d-celltypes}
{ Association test of percent pSTAT5\positive with disease status. }
{ }
\myfigure{scale=.5}{nn-pstat5-pos-t1d-celltypes}
{ Association test of NN percent pSTAT5\positive with disease status. }
{ } 
\myfigure{scale=.5}{nn-pstat5-pos-t1d-celltypes}
{ Association test of NN percent pSTAT5\positive with disease status. }
{ }

\section{Response in the whole sample}

%\begin{figure}[h]
    %\centering
    %\includegraphics[scale=.5]{IL2/figures/ungated-dose-effect.pdf}
    %\includegraphics[scale=.5]{IL2/figures/lymph-dose-effect.pdf}
    %\mycaption{figure:dose-effect}
    %{Dose effect.}
    %{
        %On the left the cumulative density function obtained from individual a on day 1 across 4 increasing doses.
        %On the right the density function.
        %The top two figures are from the ungated sample.
        %The bottom two figures are from a CD4 range gate.
        %We observe that in the unstimulated sample the distribution is already bimodal 
        %and that upon stimulation the location of the peaks does not change much but that the height changes greatly.
        %Contrary to the ungated data, the pSTAT5 distribution in the CD4 gated sample now appears unimodal when resting or stimulated
        %at the lowest 1U dose.  At the higher doses we start seeing a bimodal distribution.
        %In the ungated sample, the location of the activation peak seems to align somewhat with that of the activation peak.  }
%\end{figure}
%
%In \Cref{figure:dose-effect} we observe that at the lowest 0.1U dose there seems to be a much more stronger relative repsonse
%in the whole sample than in the lymphocyte subset which suggests that there exist other cells beside CD4\positive lymphocytes
%which are also responsive to low doses of IL-2.  
%However the relative difference in response between 10U and 1000U seems much more important in lymphocytes than in the whole sample
%suggesting that the non-CD4\positive lymphocytes in the whole sample reach saturation at 0.1U.
%Also since the resting sample sample displays a shoulder this suggests a mixture of resting and already semi-activated lymphocytes.

DILT1D and other clinical trials of IL-2 have focused on its effect on CD4\positive cells.
However, other cells may respond and this study of \emph{ex vivo} stimulated whole blood offers a chance to explore this.
Given the markers used in this study, I was interested in finding out if I could identify other cell subsets which are sensitive to proleukin.
Biologists know that any cell which carries high levels of any of the trimeric components of the IL-2 receptor, alpha, beta or gamma,
should respond to IL-2, however for pratical reasons,
due to limited number of fluorochromes,
they are not all studied as part of every flow experiment.  
%We are interested in identifying and classifying cell subsets by their sensitivity to proleukin.  
When assessing the dose-response to stimulation in a flow cytometry sample,
the classic approach is to gate cell populations in each sample based on their core markers,
and then assess the response of the functional marker in gated subset.
This approach may miss other dose-responsive cell populations which are not included in the gating.
%This might be solved by exhaustive gating
Here I will explore approaches which use the pSTAT5 response to guide the identification of cell subsets.
%While the manual gating inspired approach is the established method, it suffers from a major drawback:
%it relies on prior knowledge of the number of cell populations and consequently misses the pSTAT5 response in uncharacterised cell subsets.

\subsection{Visualising response in whole sample}

Reseachers working in low dimensional data typically begin with visualisation.
This is also important in high dimensional data but needs more innovative approaches.
I wanted to visualise the dose-response in the whole sample.
%\Gls{MDS} methods can give us an two-dimensional representation of a higher dimensional data set from a distance matrix.
Dimensionality reduction methods can give us a two-dimensional representation of a higher dimensional data set from a distance matrix.
There are well established linear ones such as \gls{PCA} but also nonlinear ones such as \gls{ISOMAP} \citep{Tenenbaum:2000jp}.
These methods are particularly popular for mass cytometry technologies such as CytoF which generate less data points but more dimensions than flow cytometry.
In this sort of dataset, more emphasis is given on uncovering cell lineages or progressions rather than discrete clusters.
Certain of these methods require computation of the complete pairwise distance matrix,
while others like PCA use the covariance matrix instead to apply a linear transform.
%Stochastic neighbour embedding attempts to preserve the proximity of objects in higher dimensional space by using probabilistic mapping of higher dimension space to lower dimension.  A drawback is that the mapping is impossible to retrieve, which is why component methods like PCA and ICA are more interpretable.  
However, information is lost when considering only the first two components of a linear projection.
Therefore, there is considerable interest in developing methods which capture both the local and global structure of the data.
%SPADE is a methods which was developed with this objective.
\acrfull{SPADE} for example represents the data as a \gls{MST}, the shortest path that connects all points in a network \citep{Simonds:2011jh}.
%The visualisation requires downsampling requires downsampling.
%A soft approach to downsampling, is to downweight points in high-density regions
I applied \gls{SPADE} on the core surface markers to construct a \gls{MST} representation, 
using all four samples from a single individual, stimulated at increasing doses of proleukin.
SPADE proceeds by first calculating the density at each data point in each FCS file.
The number of points is then reduced by preferentially removing 
high density points while preserving points of lower density,
the result of which is to ``flatten'' the density.
Once the number of points has been suffciently reduced in each FCS file, the samples are pooled and hierarchical clustering is used 
to define K=300 clusters.  These clusters are then joined using the \gls{MST} algorithm.
Once the MST has been defined, the points which were discarded in the first step are added back and assigned to their closest node.
The spanning-tree is then drawn for each FCS file with the size of the nodes representing the number of data points assigned to each node.
The tree is coloured according to the functional marker, here pSTAT5, which was not used in its construction.

However, the generated \gls{MST} requires some annotation in order to understand where the different cell types lie.
I also found that the mapping of the manual gates to the MST was not intuitive,
since cell types defined by the manual gates are spread across different branches of the tree
or overlap with different manually gated cell types (\Cref{figure:spade-celltypes}).
This makes it difficult for biologists to interpret the \gls{MST}.
Furthermore, SPADE requires pooling samples across days for construction of the \gls{MST}, which is not adviseable given that the core
markers are not expected to be stable across batches.

\begin{figure}[h]
  \centering
\begin{subfigure}[b]{.4\textwidth}
  \includegraphics[scale=.4]{IL2/figures/CB00010K-0U-2012-11-13-spade.pdf}
\caption{Resting sample.}
\end{subfigure}
\begin{subfigure}[b]{.4\textwidth}
  \includegraphics[scale=.4]{IL2/figures/CB00010K-01U-2012-11-13-spade.pdf}
  \caption{0.1 units}
\end{subfigure}
\begin{subfigure}[b]{.4\textwidth}
  \includegraphics[scale=.4]{IL2/figures/CB00010K-10U-2012-11-13-spade.pdf}
  \caption{10 units}
\end{subfigure}
\begin{subfigure}[b]{.4\textwidth}
  \includegraphics[scale=.4]{IL2/figures/CB00010K-1000U-2012-11-13-spade.pdf}
  \caption{1000 units}
\end{subfigure}
\mycaption{figure:spade-ouput}
{pSTAT5 coloured \gls{MST} generated by applying \gls{SPADE}.}
{
  The \gls{MST} was constructed from running \gls{SPADE} on the core surface markers of four samples from one individual, stimulated at increasing doses of proleukin.
  As the proleukin dose is increased more of the tree is illuminated, as the pSTAT5 MFI increases in various cell subsets.
}
\end{figure}

\begin{figure}[h]
    \centering
    \includegraphics[scale=.5]{IL2/figures/spade-celltypes.pdf}
    \mycaption{figure:spade-celltypes}
    {Mapping of cell types defined by manual gates to SPADE tree.}
    {
      Memory effectors (a), memory regulatory T cells (b), naive effectors (c) and naive regulatory T cells (d), overlap which complicates
      the interpretation.
    }
\end{figure}


\subsection{Prediction of pSTAT5 response with Random Forests}

Instead of trying to identify clusters, an alternative is to try to predict pSTAT5 response from only core marker information.
I found that using methods like \gls{RF} and \gls{MARS},
I could approximate the pSTAT5 response with a multivariate non-linear function within
the CD4\positive lymphocytes.
%This could be useful for predicting the pSTAT5 response from only core marker information.
%and contain more than N points.
%Need to give more weight to pSTAT5 in clustering.  
%It has been hypothesised that certain lymphocyte subsets are less responsive to \cytokine{IL-2} stimulation in type 1 diabetics than in healthy controls.  
%How many cell pops are they?
\gls{RF} is ran with the default parameters which grows 500 trees up to x nodes.
Each node in the tree is one the core markers which is used to split the data.
Two important summary statistics of a RF are the \gls{RSS} and the IncNodePurity parameter, a measure of variable importance in the
construction of the random forests, which are averaged over all trees in the forest.
At a node the IncNodePurity is defined as the expected \gls{RSS} decrease before and afer the split.
The IncNodePurity parameter of the RF is average overall all nodes in over all trees it the forest.
The average variance explained was \pct{40} over all \gls{RF} with some \gls{RF} explaining between \pct{20} and \pct{55} of the variance.
In \Cref{figure:IncNodePurity}, the IncNodePurity is normalised per forest so that it lies between 0 and 1, to make it comparable across forests.
Interestingly, the CD45RA marker is more important at explaining the variation in pSTAT5 response than CD25.

However, extracting a gating strategy from these models is not obvious.
Here we seek an interpretable model which can be summarised as rules to identify groups of cells which are sensitive to proleukin.
Ideally, we would like to identify cells types which are unimodal on pSTAT5 (like the naive and memory T regs) through proleukin doses,
although these cells may not necessarily be unimodal on the other markers.
%Also a RF applied on sample is not generalisable to another sample as it sensitive to small changes in the data, though less so than a CART.

\myfigure{scale=.5}
{IncNodePurity}
{
  Distribution across all samples of the normalised IncNodePurity parameter from \acrfull{RF}
}
{
  The IncNodePurity parameter is a measure of variable importance in the construction of the \acrfull{RF}.
  Interestingly, the CD45RA marker is more important than the CD25 marker in the pSTAT5 response prediction.
  This could because of the distinctly bimodal nature of the CD45RA distribution, which allows discrimination of memory cells which are high in CD25
  from naive cells, which are lower in CD25.
}


\subsection{Recursive partitioning of pSTAT5 response with regression trees} 

%This yields a baseline relative response per cell which provides additional information which can be used in the gating.  
%Here I suggest another approach of identifying responsive cells by recursive partitioning on the core marker space with a \gls{CART} on the pSTAT5 response.
One method could be to use \gls{CART} so that the variance in the pSTAT5 response is used to guide the partitioning on the core markers.
The \gls{CART} considers each core marker coordinate as a potential splitting point.
The splitting point which maximises the between class variance is selected and the data is split between the left and the right branch.
The process is applied recursively until some minimum tree node size is reached.
%the information gain of splitting at each value of that dimension.
The tree can then be pruned to a certain number of leaf nodes where each leaf node represents a partition obtained by applying the cuts
defined along the branch.
Considering the response in all cells and partitioning on side and forward scatter and subsetting the tree to the best three subsets,
I obtain the clustering in \Cref{figure:pstat5-response-decision-tree} which corresponds to the manual gating.
However trying to recreate the second step of the manual gating on side scatter and CD4 does not identify the CD4\positive cluster
but instead partitions on side scatter.
This requires some supervision to recreate the manual gating because the splits are not what the manual gating does.
Trimming the tree does not garantee that the expected clusters are returned
If left unsupervised and unpruned, the number of clusters returned is 


%Classification and regression trees are the most interpretable but risk overfitting the data and may not generalise to other samples.
%However here, overfitting is not a concern as provided cells can be identified in each sample individually which can then be checked by a human.

\begin{figure}[h]
\centering
%\includegraphics[scale=.45]{figures/pstat5-response-decision-tree-b.pdf}
\includegraphics[scale=.45]{IL2/figures/pstat5-response-decision-tree.pdf}
\mycaption{figure:pstat5-response-decision-tree}
{ \gls{CART} on side and forward scatter identifies three subsets. }
{
The regression tree obtained from recursive partitioning of side (SSCA) and forward (FSCA) scatter against the pSTAT5 response at 1000U (a).
The tree was pruned on the best three subsets.
The values at the terminal node are the expected pSTAT5 response.
Running classification tree on SSCA and FSCA, shows most of the pSTAT5 response comes from within the lymphocyte cluster (red) whereas
the black and green cluster are not as responsive.
Under these constraints, the recursive partitioning of pSTAT5 confirms that the lymphocyte cluster is the most sensitve to pSTAT5 responsive
of the other two clusters.
}
\end{figure}

\hspace{-2cm}
\begin{figure}[h]
\centering
\includegraphics[scale=.4]{figures/CB01494Y_2013-01-29.pdf}
\includegraphics[scale=.4]{figures/tree-CB01494Y_2013-01-29.pdf}
\mycaption{figure:supervised-cart}
{ Emulating the manual gating with a regression tree. }
{
  For each combination of two markers used in the manual gating, a regression tree is applied to the pSTAT5 response and the relevant split is selected
  on the marker used in manual gate.
  The values at the leaf nodes are the expected pSTAT5 response.
  Note that the tree tends to overdivide the datasets compared to the manual gating since the trees are unpruned.
  The ellipses are drawn from the mean and covariance of the clustered data.
}

\end{figure}




\subsection{Identification of highly sensitive cells with recursive partitioning using two component mixture model}

Here I suggest another approach based on the idea that by recursively identifying the highest responding cells at decreasing doses, it should be
possible to drill down to the cells which respond to the lowest dose proleukin.
%As at the highest dose of proleukin, 1000 units, the pSTAT5 distribution appears bimodal, while at lower doses the response peak is less discernable.
%In the ungated sample the majority of cells are non-responsive even at the highest dose.
As illustrated in \Cref{figure:pstat5-rpart}, cells are divided as low responders and high responders on pSTAT5 response at 1000U 
within responders, and then further divided on low/high on pSTAT5 response at 10U,
and again on pSTAT5 response at 0.1U.
Cells which consistently appear the high group are the most sensitive.
This hierarchical approach draws some similarity to the recursive partitioning using \gls{CART} except that the splitting is done using a two-component \gls{GMM}.
Also, at each step, the pSTAT5 at a lower dose is considered to discover cells which respond to the lowest dose of proleukin.

Successive univariate clustering on response is not an obvious approach to multivariate data but proves to be useful in identifying these cells.

I find that, while many of the cells identified using this approach fall within the CD4 gate,
certain highly-sensitive cells cluster in other subsets \Cref{figure:new-cell-subset}.
Excluding doublets on the basis of side scatter width,
and examining the remainder on other channels these cells appear to be monocytes (from discussion with \contributor{Marcin Pekalski} and \contributor{Tony Cutler}),
although additional markers would be required to better characterise these cells.
%Monocytes are the largest of the leukocytes and are part of the innate immun system
Importantly, these cells would have been missed by manual gating since they are not lymphocytes.


\hspace{-2cm}
\begin{figure}[h]
\centering
\includegraphics[scale=.5]{IL2/figures/pstat5-rpart.pdf}
\mycaption{figure:pstat5-rpart}
{ Recursive partitioning of pSTAT5 response into low (red) and high (green) populations to identify cells responsive to the lowest dose of proleukin. }
{
In the ungated sample the majority of cells are none responsive even at the highest dose.
Cells are divided as low responders and high responders on pSTAT5 response (i.e baseline subtracted) at 1000U 
within responders further divide on low/high on pSTAT5 response at 10U
repeat on pSTAT5 response at 0.1U
cluster ones which are consistently high, these are the most sensitive cell populations 
}
\end{figure}


\hspace{-2cm}
\begin{figure}[h]
\centering
\includegraphics[scale=.5]{IL2/figures/new-cell-subset.pdf}
\mycaption{figure:new-cell-subset}
{ Identification of low-dose sensitive non-CD4\positive lymphocyte population through recursive partitioning of pSTAT5 response. }
{
  By pooling all high-responsive cells on CD4 and CD25, we identify a cluster of cells high on CD4 (a).
  This cluster of cells is identifiable in a single sample (b).
  According to forward and side scatter, these cells are unlikely to be lymphocytes, as defined by the dashed lines (c).
  These cells show high-sensitivity to low-doses of proleukin but rapidly reach saturation (d).
  This is probably because they have high resting pSTAT5 levels.
  
}
\end{figure}




\section{Discussion}

\subsection{Comparison with previous studies}
%\subsection{Repeatability of pSTAT5}
%\subsection{Association of pSTAT5 with T1D}

The Long et al study was in a small number of individuals 66 cases and 125 controls and the reproducibility of the phenotypes was not assessed.
\contributor{Tony Cutler}, on the other hand, found that the response of the fluoresence intensity of pSTAT5 to stimulation was poorly reproducible in the various cell subsets
examined.  I attempted to improve on the reproducibility of the fluorescence intensity by using single-cell background correction and peak normalisation.
One reason why the pSTAT5 MFI is not reproducible in memory effector cells is because the pSTAT5 distribution is not unimodal.
In naive and memory regulatory T cells, the pSTAT5 distribution is unimodal but the peak shift on stimulation is not reproducible.
\contributor{Tony Cutler} claims that this is because the titration is too fine and noisy so that there is a lot of uncontrolled variation in the proleukin doses.
This motivated counting the percent of cells whose pSTAT5 fluorescence is greater than the 99th percentile of the pSTAT5 distribution in the unstimulated sample.
Since this phenotype was found to be slightly more reproducible, it was used to test the association with T1D in the four cell subsets, memory, naive, effectors and Tregs.

%They "barcode" multiple samples including 2 controls with a combination of 2 fluorescent dyes,
%pool them and stimulate in one tube with a dose of IL-2 then run through the protocol and then run on the flow.
%The samples are then deconvoluted on the flow by flourescence. I think they then "normalise" on the 2 internal controls.


%No significant associations were found which puts back into question the results of the Long et al study.

Also of concern is that the Long et al dose is too high for the studied cell subsets.
Our dose range is from 0, 0.1, 10 and 1000 U/ml, whereas theirs was 100U/ml 
We find that pSTAT5 Tregs are maximally stimulated by 10 U/ml and near maximum at 0.1U/ml,
which is in contrast to Long et al in which maximum stimulation was not achieved at 100U/ml.
One difference is that Long et al used frozen \gls{PBMC}, while we used fresh blood.
\cite{Dendrou:2009dv} showed that the repetability of a cell phenotype can be compromised with frozen samples
and this difference may explain the difference in response in the two studies.
%We had the same unit measure of IL-2 i.e U/ml.
%They also used Proleukin or equivalent.



%Tony
%I started getting some interesting data with the ND diabetics which in the end flattened to a null result.
%We initiated the LS diabetic study to see 1.
%If we could see the same findings we were initially observing in the ND diabetics and 2.
%Whether there was any relationship between IL-2 sensitivity and duration of disease.
%You could think that some events might only occur close to diagnosis.


%Such cell are of great interest because current clinical trials which administer doses of IL-2 in the hope of increasing T-reg frequency





